tests/environment.py

```diff
@@ -391,66 +400,78 @@ def launch_environment_docker(unique_id: str, skip_init: bool = False, tag: str
 class ALEntityPrplWrt(ALEntity):
     """Abstraction of ALEntity in real device."""
 
-    def __init__(self, name: str, device: None, is_controller: bool = False):
+    def __init__(self, device: None, is_controller: bool = False):
         self.device = device
-        self.name = name
-        self.bridge_name = 'br-lan'
+        self.name = device.name
 
         if is_controller:
             self.config_file_name = '/opt/prplmesh/config/beerocks_controller.conf'
         else:
             self.config_file_name = '/opt/prplmesh/config/beerocks_agent.conf'
 
-        ucc_port = self.command(("grep \"ucc_listener_port\" {} "
-                                "| cut -d'=' -f2 | cut -d\" \" -f 1").format(self.config_file_name))
-
-        device_ip_output = self.command(
-                "ip -f inet addr show {} | head -n 2".format(self.bridge_name))
-        device_ip = re.search(
-            r'inet (?P<ip>[0-9.]+)', device_ip_output.decode('utf-8')).group('ip')
-        self.log_folder = self.command(
-            "grep log_files_path {} | cut -d=\'=\' -f2".format(self.config_file_name))
-
-        ucc_socket = UCCSocket(device_ip, ucc_port)
+        ucc_port_raw = self.command("grep \"ucc_listener_port\" {}".format(self.config_file_name))
+        ucc_port = int(re.search(r'ucc_listener_port=(?P<port>[0-9]+)',
+                                 ucc_port_raw).group('port'))
+        bridge_name_raw = self.command("grep \"bridge_iface\" {}".format(self.config_file_name))
+        self.bridge_name = re.search(r'bridge_iface=(?P<bridge>.+)\r\n',
+                                     bridge_name_raw).group('bridge')
+
+        # Multiple IPs may be set to same interface. We are interested in the last one, which is set
+        # by test during board init procedure.
+        device_ip_raw = self.command(
+            "ip -family inet addr show {} | tail -n 2".format(self.bridge_name))
+        self.device_ip = re.search(r'inet (?P<ip>[0-9.]+)',
+                                   device_ip_raw).group('ip')
+
+        log_folder_raw = self.command(
+            "grep log_files_path {}".format(self.config_file_name))
+        self.log_folder = re.search(r'log_files_path=(?P<log_path>[a-zA-Z0-9_\/]+)',
+                                    log_folder_raw).group('log_path')
+
+        ucc_socket = UCCSocket(self.device_ip, int(ucc_port))
         mac = ucc_socket.dev_get_parameter('ALid')
 
         super().__init__(mac, ucc_socket, installdir, is_controller)
 
         # We always have two radios, wlan0 and wlan2
-        RadioHostapd(self, "wlan0", device=self)
-        RadioHostapd(self, "wlan2", device=self)
+        RadioHostapd(self, "wlan0")
+        RadioHostapd(self, "wlan2")
 
-    def command(self, *command: str) -> bytes:
+    def command(self, command: str) -> bytes:
         """Execute `command` in device and return its output."""
         self.device.sendline(command)
```

Comment:
So this should be
```suggestion
        self.device.sendline(" ".join(command))
```
Actually the command should be shell-quoted of course.

Also it would be better to use pexpect.replwrap.run_command, if that is available in boardfarm.

Reply:
Applied. Thank you.